oneview/resource_server_profile.go

```diff
@@ -2161,6 +2165,32 @@ func resourceServerProfileRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("template_compliance", serverProfile.TemplateCompliance)
 	d.Set("uuid", serverProfile.UUID.String())
 
+	emptyFirmware := ov.FirmwareOption{}
+	if serverProfile.Firmware != emptyFirmware {
+		firmware := make([]map[string]interface{}, 0, 1)
+		firmwareAttr := map[string]interface{}{}
+		firmwareAttr["consistency_state"] = serverProfile.Firmware.ConsistencyState
+		firmwareAttr["reapply_state"] = serverProfile.Firmware.ReapplyState
+		firmwareAttr["firmware_baseline_uri"] = serverProfile.Firmware.FirmwareBaselineUri
+		firmwareAttr["manage_firmware"] = serverProfile.Firmware.ManageFirmware
+		firmwareAttr["firmware_install_type"] = serverProfile.Firmware.FirmwareInstallType
+		firmwareAttr["firmware_activation_type"] = serverProfile.Firmware.FirmwareActivationType
+		firmwareAttr["firmware_schedule_date_time"] = serverProfile.Firmware.FirmwareScheduleDateTime
+
+		//flattening firmware to update force_install_firmware
+		if fw, ok := d.GetOk("firmware"); ok {
```

Comment:
is this mean, when firmware block is not set force_install_firmware property will be set to false.
I believe if firmware block is not set then manage_firmware also come as false. If we want to do this then spt implementation is not consistent. Need more explanation regarding the same

Reply:
This firmware block check is only for force_install_firmware attribute.. As you can see, others attributes get updated irrespective of user config. As I mentioned above, SPT doesn't have any issues with force_install_firmware param.